src/views/Home/Home.jsx

```diff
@@ -1,3 +1,32 @@
-export default function Home() {
-  return <div>This is the home view.</div>;
-}
+import React, { useState, useEffect } from 'react';
+import { getToken } from '@the-collab-lab/shopping-list-utils';
+import CreateList from '../../components/Home/CreateList';
+import JoinList from '../../components/Home/JoinList';
+
+const Home = () => {
+  const [token, setToken] = useState(localStorage.getItem('token'));
+  const [hasToken, setHasToken] = useState(token !== null);
+
+  useEffect(() => {
+    const token = localStorage.getItem('token');
+    setHasToken(token !== null);
+  }, []);
+
+  const handleCreateToken = () => {
+    setToken(localStorage.setItem('token', getToken()));
+  };
```

Comment:
While the user-facing *behavior* of the feature is correct, this code might not be doing what you expect!

On line 7, we're saying "set the `token` state to whatever is stored at the 'token' key in localStorage". On line 16, when the user clicks the `Create a list` button, we say "set the `token` state to _the  value of_ `localStorage.setItem`. This is what allows the feature to technically function correctly, but I have some food for thought:

- What is the value of calling `localStorage.setItem`? You can debug that in your console. Is it your token? If it isn't, how could you make the code more accurate?

This exercise might not be relevant as you continue building features – you may replace this code altogether, and that's fine! – but it's important to verify that your code is behaving correctly as you go!